pylot/perception/tracking/tracking_eval_operator.py

```diff
@@ -86,64 +89,55 @@ def on_watermark(self, timestamp):
-                                    or metric_name == 'mostly_lost'
-                                    or metric_name == 'partially_tracked'):
-                                ratio = metrics_summary_df[metric_name].values[
-                                    0] / metrics_summary_df[
-                                        'num_unique_objects'].values[0]
-                                self._csv_logger.info(
-                                    "{},{},{},{},{:.4f}".format(
-                                        time_epoch_ms(), sim_time,
-                                        self.config.name,
-                                        'ratio_' + metric_name, ratio))
-                            elif metric_name == 'motp':
-                                # See https://github.com/cheind/py-motmetrics/issues/92  # noqa: E501
-                                motp = (1 - metrics_summary_df[metric_name].
-                                        values[0]) * 100
-                                self._csv_logger.info(
-                                    '{},{},{},{},{:.4f}'.format(
-                                        time_epoch_ms(), sim_time,
-                                        self.config.name, metric_name, motp))
-                            elif (metric_name == 'idf1'
-                                  or metric_name == 'mota'):
-                                metric_val = metrics_summary_df[
-                                    metric_name].values[0] * 100
-                                self._csv_logger.info(
-                                    '{},{},{},{},{:.4f}'.format(
-                                        time_epoch_ms(), sim_time,
-                                        self.config.name, metric_name,
-                                        metric_val))
-                            else:
-                                self._csv_logger.info(
-                                    '{},{},{},{},{:.4f}'.format(
-                                        time_epoch_ms(), sim_time,
-                                        self.config.name, metric_name,
-                                        metrics_summary_df[metric_name].
-                                        values[0]))
-                        else:
-                            raise ValueError(
-                                'Unexpected tracking metric: {}'.format(
-                                    metric_name))
+                metrics_summary_df = self.__get_tracker_metrics(
+                    tracker_obstacles, ground_obstacles)
+                # Get runtime in ms
+                runtime = (time.time() - op_start_time) * 1000
+                self._csv_logger.info('{},{},{},{},{}'.format(
+                    time_epoch_ms(), sim_time, self.config.name, 'runtime',
+                    runtime))
+                # Write metrics to csv log file
+                self.__write_metrics_to_csv(metrics_summary_df, sim_time)
```

Comment:
Regardless of the anchoring (start or finish) we should avoid using sim_time because it's neither: if the queue gets backed up multiple (start, end) inferences will be processed and logged with the same sim_time.